## 파이썬 300제(1번 ~ 150번)
- [파이썬 300제](https://wikidocs.net/7014)

### #1. 화면에 [Hello World] 문자열을 출력하세요.

In [1]:
print("Hello World")

Hello World


### #2. 화면에 [Mary's cosmetics]을 출력하세요.

In [2]:
print("Mary's cosmetics")

Mary's cosmetics


### #3. 화면에 [신씨가 소리질렀다. "도둑이야".] 문장을 출력하세요. 

In [3]:
print('신씨가 소리질렀다. "도둑이야".')

신씨가 소리질렀다. "도둑이야".


### #4. 화면에 [C:\Windows]를 출력하세요.

In [4]:
print("C:\Windows")

C:\Windows


### #5. 다음 코드를 실행해보고 \t와 \n의 역할을 설명해보세요.
> print("안녕하세요.\n만나서\t\t반갑습니다.")

In [5]:
print("안녕하세요.\n만나서\t\t반갑습니다.")
# \n : 줄바꿈
# \t : 탭

안녕하세요.
만나서		반갑습니다.


### #6. print 함수에 두 개의 단어를 입력한 예제입니다. 아래 코드의 출력 결과를 예상해봅시다.
print ("오늘은", "일요일")

In [6]:
print ("오늘은", "일요일")
# 오늘은 일요일

오늘은 일요일


### #7. print() 함수를 사용하여 다음과 같이 출력하세요.
>naver;kakao;sk;samsung

In [8]:
print("naver",";","kakao",";","sk",";","samsung")

naver ; kakao ; sk ; samsung


In [9]:
# 정답 - sep="" 을 통해 공백 대신 ;을 표기
print("naver","kakao","sk","samsung", sep=";")

naver;kakao;sk;samsung


### #8. print() 함수를 사용하여 다음과 같이 출력하세요.

>naver/kakao/sk/samsung

In [10]:
print("naver", "kakao", "sk", "samsung", sep="/")

naver/kakao/sk/samsung


### #9. 다음 코드를 수정하여 줄바꿈이 없이 출력하세요. (힌트: end='') print 함수는 두 번 사용합니다. 세미콜론 (;)은 한줄에 여러 개의 명령을 작성하기 위해 사용합니다.

>print("first");print("second")

In [13]:
print("first", end=" ");print("second")

first second


### #10. 5/3의 결과를 화면에 출력하세요.

In [15]:
print(5/3)

1.6666666666666667


### #11. 삼성전자라는 변수로 50,000원을 바인딩해보세요. 삼성전자 주식 10주를 보유하고 있을 때 총 평가금액을 출력하세요.

In [18]:
삼성전자 = 50000
print("총평가금액:", 삼성전자 * 10)

총평가금액: 500000


### #12. 변수를 사용해서 시가총액, 현재가, PER 등을 바인딩해보세요.

In [21]:
시가총액 = 298000000000000
현재가 = 50000
PER = 15.79

In [22]:
print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))

298000000000000 <class 'int'>
50000 <class 'int'>
15.79 <class 'float'>


### #13. 변수 s와 t에는 각각 문자열이 바인딩 되어있습니다. 
> s = "hello"
> t = "python"
### 두 변수를 이용하여 [hello! python]을 출력해보세요.

In [25]:
s = "hello"
t = "python"

print(s+"! "+t)

hello! python


### #14. 아래 코드의 실행 결과를 예상해보세요.
> 2 + 2 * 3 

In [26]:
# 8
2 + 2 * 3

8

### #15. type() 함수는 데이터 타입을 판별합니다. 변수 a에는 128 숫자가 바인딩돼 있어 type 함수가 int (정수)형임을 알려줍니다.

> a = 128  
> print (type(a))  
<class 'int'>

### 아래 변수에 바인딩된 값의 타입을 판별해보세요.

> a = "132"

In [27]:
a = "132"
print(type(a))

<class 'str'>


### #16. 문자열 '720'를 정수형으로 변환해보세요.

> num_str = "720"

In [29]:
num_str = "720"
num_int = int(num_str)

print(type(num_int))

<class 'int'>


### #17. 정수 100을 문자열 '100'으로 변환해보세요.

In [30]:
num = 100
num_str = str(num)

print(type(num_str))

<class 'str'>


### #18. 문자열 "15.79"를 실수(float) 타입으로 변환해보세요.

In [31]:
num_str = "15.79"
num_float = float(num_str)

print(type(num_float))

<class 'float'>


### #19. year라는 변수가 문자열 타입의 연도를 바인딩하고 있습니다. 이를 정수로 변환한 후 최근 3년의 연도를 화면에 출력해보세요.

>year = "2020"

In [32]:
year = "2020"
year_int = int(year)
print(year_int-1, year_int-2, year_int-3)

2019 2018 2017


### #20. 에이컨이 월 48,584원에 무이자 36개월의 조건으로 홈쇼핑에서 판매되고 있습니다. 총 금액은 계산한 후 이를 화면에 출력해보세요. (변수사용하기)

In [33]:
m_cost = 48584
print("총 금액:", m_cost * 36)

총 금액: 1749024


### #21. letters가 바인딩하는 문자열에서 첫번째와 세번째 문자를 출력하세요.

> letters = 'python'

In [1]:
letters = 'python'
print(letters[0], letters[2])

p t


### #22. 자동차 번호가 다음과 같을 때 뒤에 4자리만 출력하세요.

> license_plate = "24가 2210"

In [2]:
license_plate = "24가 2210"
print(license_plate[-4:])

2210


### #23. 아래의 문자열에서 '홀' 만 출력하세요.

> string = "홀짝홀짝홀짝"

In [3]:
string = "홀짝홀짝홀짝"
print(string[::2])

홀홀홀


### #24. 문자열을 거꾸로 뒤집어 출력하세요.

> string = "PYTHON"

In [7]:
string = "PYTHON"
print(string[::-1])

NOHTYP


### #25. 아래의 전화번호에서 하이푼 ('-')을 제거하고 출력하세요.

> phone_number = "010-1111-2222"

In [9]:
# 정답 - replace 메소드를 사용하여 "-"를 " " 공백으로 바꾸어준다.
phone_number = "010-1111-2222"
phone_number_re = phone_number.replace("-", " ")
print(phone_number_re)

010 1111 2222


### #26. 25번 문제의 전화번호를 아래와 같이 모두 붙여 출력하세요.

> 01011112222

In [10]:
phone_number = "010-1111-2222"
phone_number_noblank = phone_number.replace("-", "")
print(phone_number_noblank)

01011112222


### #27. url 에 저장된 웹 페이지 주소에서 도메인을 출력하세요.

> url = "http://sharebook.kr"

In [11]:
url = "http://sharebook.kr"
print(url[-2:])

kr


In [12]:
# 정답 - .을 기준으로 split한 뒤 뒤에 덩어리를 print
url = "http://sharebook.kr"
url_split = url.split('.')
print(url_split[-1])

kr


### #28. 아래 코드의 실행 결과를 예상해보세요.

> lang = 'python'
> lang[0] = 'P'
> print(lang)

In [14]:
# lang[0] = p이므로 error가 발생한다. -> 문자열은 수정할 수 없다.
lang = 'python'
lang[0] = 'P'
print(lang)

TypeError: 'str' object does not support item assignment

### #29. 아래 문자열에서 소문자 'a'를 대문자 'A'로 변경하세요.

> string = 'abcdfe2a354a32a'

In [15]:
string = 'abcdfe2a354a32a'
string_A = string.replace("a", "A")
print(string_A)

Abcdfe2A354A32A


### #30. 아래 코드의 실행 결과를 예상해보세요.

> string = 'abcd'  
> string.replace('b', 'B')  
> print(string)

In [16]:
# abcd -> string.replace('b', 'B')에 대한 변수 설정은 따로 없었고, string에 대해 print한 것이므로 aBcd가 아닌 abcd가 나온다.
string = 'abcd'
string.replace('b', 'B')
print(string)

abcd


### #31. 아래 코드의 실행 결과를 예상해보세요.

> a = "3"  
> b = "4"  
> print(a + b)

In [17]:
# 34 -> 3과 4는 문자열이므로 3과 4가 그대로 연결되어 print 된다.
a = "3"
b = "4"
print(a + b)

34


### #32. 아래 코드의 실행 결과를 예상해보세요.

> print("Hi" * 3)

In [18]:
# HiHiHi -> Hi라는 문자열이 3번 반복해서 print 된다.
print("Hi" * 3)

HiHiHi


### #33. 화면에 '-'를 80개 출력하세요.

In [19]:
print('-' * 80)

--------------------------------------------------------------------------------


### #34. 변수에 다음과 같은 문자열이 바인딩되어 있습니다.

> t1 = 'python'  
> t2 = 'java'  

### 변수에 문자열 더하기와 문자열 곱하기를 사용해서 아래와 같이 출력해보세요.

> python java python java python java python java

In [24]:
t1 = 'python'
t2 = 'java'
t1_t2 = t1 + " " + t2 + " "
print(t1_t2 * 4)

python java python java python java python java 


### #35. 변수에 다음과 같이 문자열과 정수가 바인딩되어 있을 때 % formatting을 사용해서 다음과 같이 출력해보세요.

> name1 = "김민수"   
> age1 = 10  
> name2 = "이철희"  
> age2 = 13  
 
> 이름: 김민수 나이: 10  
> 이름: 이철희 나이: 13

In [26]:
name1 = "김민수"
age1 = 10
name2 = "이철희"
age2 = 13

print("이름: %s 나이: %d" % (name1, age1))
print("이름: %s 나이: %d" % (name2, age2))

이름: 김민수 나이: 10
이름: 이철희 나이: 13


### #36. 문자열의 format( ) 메서드를 사용해서 035번 문제를 다시 풀어보세요.

In [28]:
name1 = "김민수"
age1 = 10
name2 = "이철희"
age2 = 13

print("이름: {} 나이: {}".format(name1, age1))
print("이름: {} 나이: {}".format(name2, age2))

이름: 김민수 나이: 10
이름: 이철희 나이: 13


### #37. 파이썬 3.6부터 지원하는 f-string을 사용해서 035번 문제를 다시 풀어보세요.

In [29]:
name1 = "김민수"
age1 = 10
name2 = "이철희"
age2 = 13

print(f'이름: {name1} 나이: {age1}')
print(f'이름: {name2} 나이: {age2}')

이름: 김민수 나이: 10
이름: 이철희 나이: 13


- [% formatting, format() 함수 및 f-string 사용법](https://lcs1245.tistory.com/entry/Python-%EB%AC%B8%EC%9E%90%EC%97%B4-%ED%8F%AC%EB%A7%A4%ED%8C%85-%ED%95%98%EA%B8%B0-format-f-string)

### #38. 삼성전자의 상장주식수가 다음과 같습니다. 컴마를 제거한 후 이를 정수 타입으로 변환해보세요.

> 상장주식수 = "5,969,782,550"

In [31]:
상장주식수 = "5,969,782,550"
상장주식수_nocomma = 상장주식수.replace(",", "")
상장주식수_int = int(상장주식수_nocomma)

print(상장주식수_int, type(상장주식수_int))

5969782550 <class 'int'>


### #39. 다음과 같은 문자열에서 '2020/03'만 출력하세요.

> 분기 = "2020/03(E) (IFRS연결)"

In [34]:
분기 = "2020/03(E) (IFRS연결)"
print(분기[:7])

2020/03


### #40. 문자열의 좌우의 공백이 있을 때 이를 제거해보세요.

> data = "   삼성전자    "

In [36]:
data = "   삼성전자    "
data_re = data.replace(" ", "")
print(data_re)

삼성전자


In [37]:
# 정답 - strip() 메소드를 사용하여 모든 공백이나 특정 문자를 제거할 수 있다.
data = "   삼성전자    "
data1 = data.strip()
print(data1)

삼성전자


### #41. 다음과 같은 문자열이 있을 때 이를 대문자 BTC_KRW로 변경하세요.

> ticker = "btc_krw"

In [38]:
ticker = "btc_krw"
ticker_upper = ticker.upper()
print(ticker_upper)

BTC_KRW


### #42. 다음과 같은 문자열이 있을 때 이를 소문자 btc_krw로 변경하세요.

> ticker = "BTC_KRW"

In [39]:
ticker = "BTC_KRW"
ticker_lower = ticker.lower()
print(ticker_lower)

btc_krw


### #43. 문자열 'hello'가 있을 때 이를 'Hello'로 변경해보세요.

In [40]:
str = "hello"
str_c = str.capitalize()
print(str_c)

Hello


### #44. 파일 이름이 문자열로 저장되어 있을 때 endswith 메서드를 사용해서 파일 이름이 'xlsx'로 끝나는지 확인해보세요.

> file_name = "보고서.xlsx"

In [41]:
file_name = "보고서.xlsx"
file_name.endswith("xlsx")

True

### #45. 파일 이름이 문자열로 저장되어 있을 때 endswith 메서드를 사용해서 파일 이름이 'xlsx' 또는 'xls'로 끝나는지 확인해보세요.

> file_name = "보고서.xlsx"

In [43]:
file_name = "보고서.xlsx"
file_name.endswith(("xlsx", "xls"))
# endswith 메서드를 사용하면서 확인해야 할 문자가 두 개 이상인 경우 ()로 묶어서 사용한다.

True

### #46. 파일 이름이 문자열로 저장되어 있을 때 startswith 메서드를 사용해서 파일 이름이 '2020'로 시작하는지 확인해보세요.

> file_name = "2020_보고서.xlsx"

In [44]:
file_name = "2020_보고서.xlsx"
file_name.startswith("2020")

True

### #47. 다음과 같은 문자열이 있을 때 공백을 기준으로 문자열을 나눠보세요.

> a = "hello world"

In [46]:
a = "hello world"
a.split()

['hello', 'world']

### #48. 다음과 같이 문자열이 있을 때 btc와 krw로 나눠보세요.

> ticker = "btc_krw"

In [47]:
ticker = "btc_krw"
ticker.split("_")

['btc', 'krw']

### #49. 다음과 같이 날짜를 표현하는 문자열이 있을 때 연도, 월, 일로 나눠보세요.

> date = "2020-05-01"

In [48]:
date = "2020-05-01"
date.split('-')

['2020', '05', '01']

### #50. 문자열의 오른쪽에 공백이 있을 때 이를 제거해보세요.

In [49]:
data = "039490     "
data.rstrip()

'039490'

- [strip(), rstrip(), lstrip()](https://106hht.tistory.com/51)

----------

## 리스트 관련 문법


In [6]:
array = [i for i in range(10)]
array

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

In [8]:
array1 = [i for i in range(20) if i % 2 == 1]
array1

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19]

In [9]:
# 위의 식은 다음과 같이 사용된 것을 줄여서 사용할 수 있다.
array2 = []
for i in range(20):
    if i % 2 == 1:
        array2.append(i)
array2

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19]

In [11]:
# M x N 크기의 2차원 리스트
m = 3
n = 4

array3 = [[0]*m for _ in range(n)]
array3 # 3 x 4 크기의 matrix가 만들어진 것

[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0]]

In [12]:
# 주의! M x N 크기의 2차원 리스트를 다음과 같이 쓰면 안된다.
array4 = [[0]*m]*n
array4 # 똑같은 [0, 0, 0] 4개를 만든 것에 불과

[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0]]

#### 리스트 관련 메서드
- append() : 원소 1개 삽입
- sort() / sort(reverse=True) : 오름차순 정렬 / 내림차순 정렬
- reverse() : 순서 반대로 돌리기
- insert(위치, 특정값) : 원하는 위치에 특정값 삽입
- count() : 특정 값의 개수
- remove() : 특정 값 제거. 그 값이 여러 개이면 맨 앞 숫자 하나만 제거된다.

In [13]:
# remove()
a = [1, 2, 3, 4, 5, 5, 5]
a.remove(5)
print("a =", a)

# 특정 값을 모두 제거하고 싶을 때
b = [1, 2, 3, 4, 5, 5, 5]
remove_set = {3, 5}
result = [i for i in b if i not in remove_set]
print("b =", result)

a = [1, 2, 3, 4, 5, 5]
b = [1, 2, 4]


-----------

### #51. 2016년 11월 영화 예매 순위 기준 top3는 다음과 같습니다. 영화 제목을 movie_rank 이름의 리스트에 저장해보세요. (순위 정보는 저장하지 않습니다.)

>1위	닥터 스트레인지  
2위	스플릿  
3위	럭키  

In [1]:
movie_rank = ["닥터 스트레인지", "스플릿", "럭키"]
movie_rank

['닥터 스트레인지', '스플릿', '럭키']

### #52. 051의 movie_rank 리스트에 "배트맨"을 추가하라.

In [2]:
movie_rank.append("배트맨")
movie_rank

['닥터 스트레인지', '스플릿', '럭키', '배트맨']

### #53. movie_rank 리스트에는 아래와 같이 네 개의 영화 제목이 바인딩되어 있다. "슈퍼맨"을 "닥터 스트레인지"와 "스플릿" 사이에 추가하라.

> movie_rank = ['닥터 스트레인지', '스플릿', '럭키', '배트맨']

In [4]:
movie_rank = ['닥터 스트레인지', '스플릿', '럭키', '배트맨']
movie_rank.insert(1, '슈퍼맨')
movie_rank

['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']

### #54. movie_rank 리스트에서 '럭키'를 삭제하라.

> movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']

In [14]:
movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '럭키', '배트맨']
movie_rank.remove('럭키')
movie_rank

['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']

### #55. movie_rank 리스트에서 '스플릿' 과 '배트맨'을 를 삭제하라.

> movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']

In [16]:
movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']
movie_rank.remove('스플릿')
movie_rank.remove('배트맨')
movie_rank

['닥터 스트레인지', '슈퍼맨']

In [17]:
# 정답 - del과 리스트 인덱싱을 활용하여 삭제하는 방법도 있다.
movie_rank = ['닥터 스트레인지', '슈퍼맨', '스플릿', '배트맨']
del movie_rank[2]
del movie_rank[2]
print(movie_rank)

['닥터 스트레인지', '슈퍼맨']


### #56. lang1과 lang2 리스트가 있을 때 lang1과 lang2의 원소를 모두 갖고 있는 langs 리스트를 만들어라.

> lang1 = ["C", "C++", "JAVA"]  
 lang2 = ["Python", "Go", "C#"]  
 
> langs = ['C', 'C++', 'JAVA', 'Python', 'Go', 'C#']

In [19]:
lang1 = ["C", "C++", "JAVA"]
lang2 = ["Python", "Go", "C#"]

print("langs =", lang1 + lang2)

langs = ['C', 'C++', 'JAVA', 'Python', 'Go', 'C#']


### #57. 다음 리스트에서 최댓값과 최솟값을 출력하라. (힌트: min(), max() 함수 사용)

> nums = [1, 2, 3, 4, 5, 6, 7]

In [23]:
nums = [1, 2, 3, 4, 5, 6, 7]
print("min:", min(nums))
print("max:", max(nums))

min: 1
max: 7


### #58. 다음 리스트의 합을 출력하라.

> nums = [1, 2, 3, 4, 5]

In [26]:
nums = [1, 2, 3, 4, 5]
a = 0

for i in nums:
    a += i
print(a)

15


In [27]:
# 정답 - sum() 함수 사용하기
nums = [1, 2, 3, 4, 5]
print(sum(nums))

15


### #59. 다음 리스트에 저장된 데이터의 개수를 화면에 구하하라.

> cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]

In [29]:
cook = ["피자", "김밥", "만두", "양념치킨", "족발", "피자", "김치만두", "쫄면", "소시지", "라면", "팥빙수", "김치전"]
print(len(cook))

12


### #60. 다음 리스트의 평균을 출력하라.

> nums = [1, 2, 3, 4, 5]

In [30]:
nums = [1, 2, 3, 4, 5]
print(sum(nums) / len(nums))

3.0


### #61. price 변수에는 날짜와 종가 정보가 저장돼 있다. 날짜 정보를 제외하고 가격 정보만을 출력하라.

> price = ['20180728', 100, 130, 140, 150, 160, 170]

In [31]:
price = ['20180728', 100, 130, 140, 150, 160, 170]
print(price[1:])

[100, 130, 140, 150, 160, 170]


### #62. 슬라이싱을 사용해서 홀수만 출력하라.

> nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [32]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(nums[::2])

[1, 3, 5, 7, 9]


### #63. 슬라이싱을 사용해서 짝수만 출력하라.

> nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [33]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(nums[1::2])

[2, 4, 6, 8, 10]


### #64. 슬라이싱을 사용해서 리스트의 숫자를 역 방향으로 출력하라.

> nums = [1, 2, 3, 4, 5]

In [34]:
nums = [1, 2, 3, 4, 5]
print(nums[::-1])

[5, 4, 3, 2, 1]


### #65. interest 리스트에는 아래의 데이터가 바인딩되어 있다.

>interest = ['삼성전자', 'LG전자', 'Naver']

### interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

> 삼성전자 Naver

In [35]:
interest = ['삼성전자', 'LG전자', 'Naver']
print(interest[0], interest[2])

삼성전자 Naver


### #66. interest 리스트에는 아래의 데이터가 바인딩되어 있다.

> interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']

### interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

> 삼성전자 LG전자 Naver SK하이닉스 미래에셋대우

In [37]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print(" ".join(interest))

삼성전자 LG전자 Naver SK하이닉스 미래에셋대우


- [join() 함수 사용법](https://blockdmask.tistory.com/468)

### #67. interest 리스트에는 아래의 데이터가 바인딩되어 있다.

> interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']

### interest 리스트를 사용하여 아래와 같이 화면에 출력하라.

> 삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우

In [38]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print("/".join(interest))

삼성전자/LG전자/Naver/SK하이닉스/미래에셋대우


### #68. interest 리스트에는 아래의 데이터가 바인딩되어 있다.

>interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
### join() 메서드를 사용해서 interest 리스트를 아래와 같이 화면에 출력하라.
>삼성전자  
LG전자  
Naver  
SK하이닉스  
미래에셋대우

In [39]:
interest = ['삼성전자', 'LG전자', 'Naver', 'SK하이닉스', '미래에셋대우']
print("\n".join(interest))

삼성전자
LG전자
Naver
SK하이닉스
미래에셋대우


### #69. 회사 이름이 슬래시 ('/')로 구분되어 하나의 문자열로 저장되어 있다.

> string = "삼성전자/LG전자/Naver"
### 이를 interest 이름의 리스트로 분리 저장하라.

> print(interest)  
> ['삼성전자', 'LG전자', 'Naver']

In [43]:
string = "삼성전자/LG전자/Naver"

interest = string.split("/")
print(interest)

['삼성전자', 'LG전자', 'Naver']


### #70. 리스트에 있는 값을 오름차순으로 정렬하세요.

> data = [2, 4, 3, 1, 5, 10, 9]

In [47]:
# sort() 함수 사용하는 경우
data = [2, 4, 3, 1, 5, 10, 9]
data.sort()
print(data)

[1, 2, 3, 4, 5, 9, 10]


In [48]:
# sorted() 함수 사용하는 경우
data = [2, 4, 3, 1, 5, 10, 9]
data2 = sorted(data)
print(data2)

[1, 2, 3, 4, 5, 9, 10]


### #71. my_variable 이름의 비어있는 튜플을 만들라.

In [50]:
my_variable = ()
print(type(my_variable))

<class 'tuple'>


### #72. 2016년 11월 영화 예매 순위 기준 top3는 다음과 같다. 영화 제목을 movie_rank 이름의 튜플에 저장하라. (순위 정보는 저장하지 않는다.)
> 1위 	닥터 스트레인지  
2위	스플릿  
3위	럭키

In [51]:
movie_rank = ('닥터 스트레인지', '스플릿', '럭키')
print(movie_rank, type(movie_rank))

('닥터 스트레인지', '스플릿', '럭키') <class 'tuple'>


### #73. 숫자 1 이 저장된 튜플을 생성하라.

In [54]:
tuple = (1, ) # 하나의 데이터가 저장되는 경우, 쉼표를 입력해야 한다.
type(tuple)

tuple

In [55]:
# cf. 쉼표를 입력하지 않는 경우 
tuple2 = (1)
type(tuple2)

int

### #74. 다음 코드를 실행해보고 오류가 발생하는 원인을 설명하라.

In [56]:
t = (1, 2, 3)
t[0] = 'a'
# 튜플은 원소(element)의 값을 변경하는 것이 불가능하기 때문에 t[0] = 1을 a로 바꿀 수 없다.

TypeError: 'tuple' object does not support item assignment

### #75. 아래와 같이 t에는 1, 2, 3, 4 데이터가 바인딩되어 있다. t가 바인딩하는 데이터 타입은 무엇인가?

> t = 1, 2, 3, 4

In [57]:
t = 1, 2, 3, 4
type(t)
# 원칙적으로 튜플은 괄호와 함께 데이터를 정의해야 하지만, 사용자 편의를 위해 괄호 없이도 동작한다.

tuple

### #76. 변수 t에는 아래와 같은 값이 저장되어 있다. 변수 t가 ('A', 'b', 'c') 튜플을 가리키도록 수정 하라.

> t = ('a', 'b', 'c')

In [58]:
t = ('a', 'b', 'c')
t = ('A', 'b', 'c') # 튜플의 값은 변경할 수 없으므로, 새로운 튜플을 만들고 t 라는 변수를 업데이트 해야 한다.
t

('A', 'b', 'c')

### #77. 다음 튜플을 리스트로 변환하라.

> interest = ('삼성전자', 'LG전자', 'SK Hynix')

In [59]:
interest = ('삼성전자', 'LG전자', 'SK Hynix')
interest_list = list(interest)
interest_list

['삼성전자', 'LG전자', 'SK Hynix']

### #78. 다음 리스트를 튜플로 변경하라.

> interest = ['삼성전자', 'LG전자', 'SK Hynix']

In [1]:
interest = ['삼성전자', 'LG전자', 'SK Hynix']
interest_tuple = tuple(interest)
interest_tuple

('삼성전자', 'LG전자', 'SK Hynix')

### #79. 다음 코드의 실행 결과를 예상하라.

>temp = ('apple', 'banana', 'cake')  
a, b, c = temp  
print(a, b, c)

In [2]:
temp = ('apple', 'banana', 'cake')
a, b, c = temp
print(a, b, c)

apple banana cake


### #80. 1 부터 99까지의 정수 중 짝수만 저장된 튜플을 생성하라.

> (2, 4, 6, 8 ... 98)

In [6]:
t = tuple(range(2,100,2))
print(t)

(2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98)


### #81. 다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 좌측 8개의 값을 valid_score 변수에 바인딩하여라.

> scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [7]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
*valid_score, _, _ = scores
valid_score

[8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5]

In [8]:
# * 사용방법
a, b, *c = (0, 1, 2, 3, 4, 5)
print(a)
print(b)
print(c)

0
1
[2, 3, 4, 5]


### #82. 다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 우측 8개의 값을 valid_score 변수에 바인딩하여라.

> scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [9]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
_, _, *valid_score = scores
valid_score

[8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

### #83. 다음과 같이 10개의 값이 저장된 scores 리스트가 있을 때, start expression을 사용하여 가운데 있는 8개의 값을 valid_score 변수에 바인딩하여라.

> scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]

In [10]:
scores = [8.8, 8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8, 9.4]
_, *valid_score, _ = scores
valid_score

[8.9, 8.7, 9.2, 9.3, 9.7, 9.9, 9.5, 7.8]

### #84. temp 이름의 비어있는 딕셔너리를 만들라.

In [12]:
temp = { }
print(type(temp))

<class 'dict'>


### #85. 다음 아이스크림 이름과 희망 가격을 딕셔너리로 구성하라.

> 이름 / 희망 가격  
 메로나 / 1000  
 폴라포 / 1200  
 빵빠레 / 1800  

In [14]:
ice_cream = {'메로나': 1000, '폴라포': 1200, '빵빠레': 1800}
print(ice_cream)

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800}


### #86. 085 번의 딕셔너리에 아래 아이스크림 가격정보를 추가하라.

>이름 / 희망 가격  
죠스바 / 1200  
월드콘 / 1500

In [15]:
ice_cream = {'메로나': 1000, '폴라포': 1200, '빵빠레': 1800}
ice_cream['죠스바'] = 1200
ice_cream['월드콘'] = 1500
print(ice_cream)

{'메로나': 1000, '폴라포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


### #87. 다음 딕셔너리를 사용하여 메로나 가격을 출력하라.

> ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

In [18]:
ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

print("메로나 가격:", ice['메로나'])

메로나 가격: 1000


### #88. 다음 딕셔너리에서 메로나의 가격을 1300으로 수정하라.

> ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

In [19]:
ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

ice['메로나'] = 1300
print(ice)

{'메로나': 1300, '폴로포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


### #89. 다음 딕셔너리에서 메로나를 삭제하라.

> ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}

In [20]:
ice = {'메로나': 1000,
       '폴로포': 1200,
       '빵빠레': 1800,
       '죠스바': 1200,
       '월드콘': 1500}
del ice['메로나']
print(ice)

{'폴로포': 1200, '빵빠레': 1800, '죠스바': 1200, '월드콘': 1500}


### #90. 다음 코드에서 에러가 발생한 원인을 설명하라.

In [21]:
icecream = {'폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
icecream['누가바']
# icecream이라는 dict에는 '누가바'라는 key가 존재하지 않기 때문에 error가 발생(딕셔너리에 없는 키를 사용해서 인덱싱하면 에러가 발생)

KeyError: '누가바'

### #91. 아래의 표에서, 아이스크림 이름을 키값으로, (가격, 재고) 리스트를 딕셔너리의 값으로 저장하라. 딕셔너리의 이름은 inventory로 한다.

>이름 / 가격 / 재고  
메로나 / 300 / 20  
비비빅 / 400 / 3  
죠스바 / 250 / 100  

In [22]:
inventory = {'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100]}
print(inventory)

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100]}


### #92. inventory 딕셔너리에서 메로나의 가격을 화면에 출력하라.

In [24]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}

print(inventory["메로나"][0], "원")

300 원


### #93. inventory 딕셔너리에서 메로나의 재고를 화면에 출력하라.

In [25]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}

print(inventory["메로나"][1], "개")

20 개


### #94. inventory 딕셔너리에 아래 데이터를 추가하라.
>이름 / 가격 / 재고  
월드콘 / 500 / 7

In [26]:
inventory = {"메로나": [300, 20],
              "비비빅": [400, 3],
              "죠스바": [250, 100]}
inventory["월드콘"] = [500, 7]
print(inventory)

{'메로나': [300, 20], '비비빅': [400, 3], '죠스바': [250, 100], '월드콘': [500, 7]}


### #95. 다음의 딕셔너리로부터 key 값으로만 구성된 리스트를 생성하라.

> icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [28]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
print(list(icecream.keys()))

['탱크보이', '폴라포', '빵빠레', '월드콘', '메로나']


### #96. 다음의 딕셔너리에서 values 값으로만 구성된 리스트를 생성하라.

> icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [29]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
print(list(icecream.values()))

[1200, 1200, 1800, 1500, 1000]


### #97. icecream 딕셔너리에서 아이스크림 판매 금액의 총합을 출력하라.

> icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}

In [30]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
print(sum(icecream.values()))

6700


### #98. 아래의 new_product 딕셔너리를 다음 icecream 딕셔너리에 추가하라.

>icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}  
new_product = {'팥빙수':2700, '아맛나':1000}

In [32]:
icecream = {'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000}
new_product = {'팥빙수':2700, '아맛나':1000}
icecream.update(new_product)
print(icecream)

{'탱크보이': 1200, '폴라포': 1200, '빵빠레': 1800, '월드콘': 1500, '메로나': 1000, '팥빙수': 2700, '아맛나': 1000}


### #99. 아래 두 개의 튜플을 하나의 딕셔너리로 변환하라. keys를 키로, vals를 값으로 result 이름의 딕셔너리로 저장한다.

>keys = ("apple", "pear", "peach")  
vals = (300, 250, 400)

In [34]:
keys = ("apple", "pear", "peach")
vals = (300, 250, 400)
result = dict(zip(keys, vals))
print(result)

{'apple': 300, 'pear': 250, 'peach': 400}


### #100. date와 close_price 두 개의 리스트를 close_table 이름의 딕셔너리로 생성하라.

> date = ['09/05', '09/06', '09/07', '09/08', '09/09']  
close_price = [10500, 10300, 10100, 10800, 11000]

In [35]:
date = ['09/05', '09/06', '09/07', '09/08', '09/09']
close_price = [10500, 10300, 10100, 10800, 11000]
close_table = dict(zip(date, close_price))
print(close_table)

{'09/05': 10500, '09/06': 10300, '09/07': 10100, '09/08': 10800, '09/09': 11000}


### #101. 파이썬에서 True 혹은 False를 갖는 데이터 타입은 무엇인가?
정답 : bool 타입

### #102. 아래 코드의 출력 결과를 예상하라.

> print(3 == 5)

In [36]:
print(3 == 5)
# False

False


### #103. 아래 코드의 출력 결과를 예상하라

> print(3 < 5)

In [37]:
print(3 < 5)
# True

True


### #104. 아래 코드의 결과를 예상하라.

> x = 4  
print(1 < x < 5)

In [38]:
x = 4
print(1 < x < 5)
# True

True


### #105. 아래 코드의 결과를 예상하라.

> print ((3 == 3) and (4 != 3))

In [39]:
print ((3 == 3) and (4 != 3))
# True

True


### #106. 아래 코드에서 에러가 발생하는 원인에 대해 설명하라.

> print(3 => 4)

In [40]:
print(3 => 4)
# =>와 같은 연산자가 존재하지 않는다. 크거나 같은지 확인하려면 >=을 사용해야 한다.

SyntaxError: expression cannot contain assignment, perhaps you meant "=="? (3612698823.py, line 1)

### #107. 아래 코드의 출력 결과를 예상하라

> if 4 < 3:  
    print("Hello World")

In [42]:
if 4 < 3:
    print("Hello World")
# 아무것도 출력되지 않는다.

### #108. 아래 코드의 출력 결과를 예상하라

>if 4 < 3:  
    print("Hello World.")  
else:  
    print("Hi, there.")  

In [43]:
if 4 < 3:
    print("Hello World.")
else:
    print("Hi, there.")
# Hi, there. 가 출력된다.

Hi, there.


### #109. 아래 코드의 출력 결과를 예상하라

>if True :  
    print ("1")  
    print ("2")  
else :  
    print("3")  
print("4")

In [44]:
if True :
    print ("1")
    print ("2")
else :
    print("3")
print("4")
# 1
# 2
# 4

1
2
4


### #110. 아래 코드의 출력 결과를 예상하라

>if True :  
    if False:  
        print("1")  
        print("2")  
    else:  
        print("3")  
else :  
    print("4")  
print("5")

In [45]:
if True :
    if False:
        print("1")
        print("2")
    else:
        print("3")
else :
    print("4")
print("5")
# 3
# 5

3
5


### #111. 사용자로부터 입력받은 문자열을 두 번 출력하라.

In [1]:
user = input()
print(user * 2)

안녕하세요
안녕하세요안녕하세요


### #112. 사용자로부터 하나의 숫자를 입력받고, 입력 받은 숫자에 10을 더해 출력하라.

In [3]:
user = int(input("숫자를 입력하세요: "))
print(user + 10)

숫자를 입력하세요: 30
40


### #113. 사용자로부터 하나의 숫자를 입력 받고 짝수/홀수를 판별하라.

In [4]:
user = int(input("숫자를 입력하세요: "))
if user % 2 == 0:
    print("짝수")
else:
    print("홀수")

숫자를 입력하세요: 30
짝수


### #114. 사용자로부터 값을 입력받은 후 해당 값에 20을 더한 값을 출력하라. 단 사용자가 입력한 값과 20을 더한 계산 값이 255를 초과하는 경우 255를 출력해야 한다.

In [5]:
user = int(input("입력: "))
if user + 20 <= 255:
    print(user + 20)
else:
    print(255)

입력: 200
220


In [6]:
user = int(input("입력: "))
if user + 20 <= 255:
    print(user + 20)
else:
    print(255)

입력: 240
255


### #115. 사용자로부터 하나의 값을 입력받은 후 해당 값에 20을 뺀 값을 출력하라. 단 출력 값의 범위는 0~255이다. 예를 들어 결괏값이 0보다 작은 값이되는 경우 0을 출력하고 255보다 큰 값이 되는 경우 255를 출력해야 한다.

In [7]:
user = int(input("입력: "))
num = user - 20

if num < 0:
    print(0)
elif num <= 255:
    print(num)
else:
    print(255)

입력: 200
180


In [8]:
user = int(input("입력: "))
num = user - 20

if num < 0:
    print(0)
elif num <= 255:
    print(num)
else:
    print(255)

입력: 15
0


### #116. 사용자로부터 입력 받은 시간이 정각인지 판별하라.

In [10]:
user = input("입력: ")
time = user[-2:]

if time == "00":
    print("정각 입니다.")
else:
    print("정각이 아닙니다.")

입력: 02:00
정각 입니다.


### #117. 사용자로 입력받은 단어가 아래 fruit 리스트에 포함되어 있는지를 확인하라. 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.
> fruit = ["사과", "포도", "홍시"]

In [11]:
fruit = ["사과", "포도", "홍시"]

user = input("좋아하는 과일은? ")
if user in fruit:
    print("정답입니다.")
else: 
    print("오답입니다.")

좋아하는 과일은? 사과
정답입니다.


### #118. 투자 경고 종목 리스트가 있을 때 사용자로부터 종목명을 입력 받은 후 해당 종목이 투자 경고 종목이라면 '투자 경고 종목입니다'를 아니면 "투자 경고 종목이 아닙니다."를 출력하는 프로그램을 작성하라.
> warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]

In [12]:
warn_investment_list = ["Microsoft", "Google", "Naver", "Kakao", "SAMSUNG", "LG"]

user = input("입력: ")
if user in warn_investment_list:
    print("투자 경고 종목입니다.")
else:
    print("투자 경고 종목이 아닙니다.")

입력: Google
투자 경고 종목입니다.


### #119. 아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 키 (key) 값에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

> fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [15]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

user = input("제가좋아하는계절은: ")
if user in fruit.keys():
    print("정답입니다.")
else:
    print("오답입니다.")

제가좋아하는계절은: 봄
정답입니다.


### #120. 아래와 같이 fruit 딕셔너리가 정의되어 있다. 사용자가 입력한 값이 딕셔너리 값 (value)에 포함되었다면 "정답입니다"를 아닐 경우 "오답입니다" 출력하라.

> fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

In [16]:
fruit = {"봄" : "딸기", "여름" : "토마토", "가을" : "사과"}

user = input("좋아하는과일은? ")
if user in fruit.values():
    print("정답입니다.")
else:
    print("오답입니다.")

좋아하는과일은? 한라봉
오답입니다.


### #121. 사용자로부터 문자 한 개를 입력 받고, 소문자일 경우 대문자로, 대문자 일 경우, 소문자로 변경해서 출력하라.

In [20]:
user = input("입력: ")
if user.islower() == True:
    print(user.upper())
else:
    print(user.lower())

입력: a
A


### #122. 점수 구간에 해당하는 학점이 아래와 같이 정의되어 있다. 사용자로부터 score를 입력받아 학점을 출력하라.

> 점수 / 학점  
81\~100 / A  
61\~80	/ B  
41\~60	/ C  
21\~40	/ D  
0\~20	/ E  

In [21]:
user = int(input("score: "))
if user <= 100 and user > 80:
    print("grade is A")
elif user > 60:
    print("grade is B")
elif user > 40:
    print("grade is C")
elif user > 20:
    print("grade is D")
else:
    print("grade is E")

score: 83
grade is A


### #123. 사용자로부터 달러, 엔, 유로, 또는 위안 금액을 입력받은 후 이를 원으로 변환하는 프로그램을 작성하라. 각 통화별 환율은 다음과 같다. 사용자는 100 달러, 1000 엔, 13 유로, 100 위안과 같이 금액과 통화명 사이에 공백을 넣어 입력한다고 가정한다.

>통화명 / 환율  
달러 / 1167  
엔 / 1.096  
유로 / 1268  
위안 / 171

In [5]:
user = input("입력: ")
num, currency = user.split()
if currency == "달러":
    print(float(num) * 1167, "원")
elif currency == "엔":
    print(float(num) * 1.096, "원")
elif currency == "유로":
    print(float(num) * 1268, "원")
else:
    print(float(num) * 171, "원")

입력: 100 달러
116700.0 원


In [6]:
# 정답 - 딕셔너리에 원소 추가만 해주면 바로 사용할 수 있다는 장점이 있다.
환율 = {"달러": 1167, 
        "엔": 1.096, 
        "유로": 1268, 
        "위안": 171}
user = input("입력: ")
num, currency = user.split()
print(float(num) * 환율[currency], "원") # 환율[currency] = split()을 했을 때 뒷부분에 해당하는 화폐에 대한 value 값

입력: 100 달러
116700.0 원


### #124. 사용자로부터 세 개의 숫자를 입력 받은 후 가장 큰 숫자를 출력하라.

> input number1: 10  
> input number2: 9  
> input number3: 20  
> -> 20

In [8]:
n1 = int(input("number1: "))
n2 = int(input("number2: "))
n3 = int(input("number3: "))

if n1 >= n2 and n1 >= n3:
    print(n1)
elif n2 >= n1 and n2 >= n3:
    print(n2)
else: 
    print(n3)

number1: 10
number2: 9
number3: 20
20


### #125. 휴대폰 번호 앞자리에 따라 통신사는 아래와 같이 구분된다. 사용자로부터 휴대전화 번호를 입력 받고, 통신사를 출력하는 프로그램을 작성하라.

>번호 / 통신사  
011	/ SKT  
016	/ KT  
019	/ LGU  
010 / 알수없음

In [9]:
user = input("휴대전화 번호 입력: ")
num = user[:3]

if num == "011":
    print("당신은 SKT 사용자입니다.")
elif num == "016":
    print("당신은 KT 사용자입니다.")
elif num == "019":
    print("당신은 LGU 사용자입니다.")
else:
    print("당신은 알수없음 사용자입니다.")

휴대전화 번호 입력: 011-345-1922
당신은 SKT 사용자입니다.


In [10]:
# 정답 - f-string을 사용하는 방법도 존재
number = input("휴대전화 번호 입력: ")
num = number.split("-")[0]
if num == "011":
    com = "SKT"
elif num == "016":
    com = "KT"
elif num == "019":
    com = "LGU"
else:
    com = "알수없음"
print(f"당신은 {com} 사용자입니다.")

휴대전화 번호 입력: 011-345-1922
당신은 SKT 사용자입니다.


### #126. 우편번호는 5자리로 구성되는데, 앞의 세자리는 구를 나타낸다. 예를들어, 강북구의 경우 010, 011, 012 세 자리로 시작한다. 사용자로 부터 5자리 우편번호를 입력받고 구를 판별하라.
>01 0 강북구 / 1 강북구 / 2 강북구 / 3 도봉구 / 4 도봉구 / 5 도봉구 / 6 노원구 / 7 노원구 / 8 노원구 / 9 노원구

In [11]:
user = input("우편번호: ")
num = int(user[2])
if num == 0 or num == 1 or num == 2:
    print("강북구")
elif num == 3 or num == 4 or num == 5:
    print("도봉구")
else:
    print("노원구")

우편번호: 01400
도봉구


In [12]:
# 정답 - 리스트에 있는지 여부로도 판단할 수 있다.
우편번호 = input("우편번호: ")
우편번호 = 우편번호[:3]
if 우편번호 in ["010", "011", "012"]:
    print("강북구")
elif 우편번호 in ["014", "015", "016"]:
    print("도봉구")
else:
    print("노원구")

우편번호: 01400
도봉구


### #127. 주민등록번호 뒷 자리 7자리 중 첫째 자리는 성별을 나타내는데, 1, 3은 남자 2, 4는 여자를 의미한다. 사용자로부터 13자리의 주민등록번호를 입력 받은 후 성별 (남자, 여자)를 출력하는 프로그램을 작성하라.

In [13]:
user = input("주민등록번호: ")
sex = user[7]

if sex in ["1", "3"]:
    print("남자")
elif sex in ["2", "4"]:
    print("여자")

주민등록번호: 821010-1635210
남자


### #128. 주민등록번호의 뒷 자리 7자리 중 두번째와 세번째는 지역코드를 의미한다. 주민 등록 번호를 입력 받은 후 출생지가 서울인지 아닌지 판단하는 코드를 작성하라

> 지역코드 / 출생지  
00 \~ 08 / 서울  
09 \~ 12 / 부산

In [14]:
user = input("주민등록번호:")
area = user[8:10]

if area in ["00", "01", "02", "03", "04", "05", "06", "07", "08"]:
    print("서울 입니다.")
else:
    print("서울이 아닙니다.")

주민등록번호:821010-1635210
서울이 아닙니다.


In [15]:
user = input("주민등록번호:")
area = user[8:10]

if area in ["00", "01", "02", "03", "04", "05", "06", "07", "08"]:
    print("서울 입니다.")
else:
    print("서울이 아닙니다.")

주민등록번호:861010-1015210
서울 입니다.


### #129. 주민등록번호는 13자리로 구성되는데 마지막 자리수는 주민등록번호의 유효성을 체크하는데 사용된다. 먼저 앞에서부터 12자리의 숫자에 2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5를 차례로 곱한 뒤 그 값을 전부 더한다. 연산 결과 값을 11로 나누면 나머지가 나오는데 11에서 나머지를 뺀 값이 주민등록번호의 마지막 번호가 된다. 
### 821010-1635210에 대해서 계산을 해보면 마지막 자리는 4가 되어야 함을 알 수 있다. 즉, 821010-1635210은 유효하지 않은 주민등록번호임을 알 수 있다.
### 사용자로부터 주민등록번호를 입력받은 후 주민등록번호가 유효한지를 출력하는 프로그램을 작성하라.

In [17]:
user = input("주민등록번호:")
num1 = int(user[0]) * 2 + int(user[1]) * 3 + int(user[2]) * 4 + int(user[3]) * 5 + int(user[4]) * 6 + \
        int(user[5]) * 7 + int(user[7]) * 8 + int(user[8]) * 9 + int(user[9]) * 2 + int(user[10]) * 3 + \
        int(user[11]) * 4 + int(user[12]) * 5
num2 = num1 % 11

if 11 - num2 == int(user[13]):
    print("유효한 주민등록번호입니다.")
else:
    print("유효하지 않은 주민등록번호입니다.")

주민등록번호:821010-1635210
유효하지 않은 주민등록번호입니다.


### #130. 아래 코드는 비트코인의 가격 정보를 딕셔너리로 가져오는 코드이다.
> import requests  
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']

### btc 딕셔너리 안에는 시가, 종가, 최고가, 최저가 등이 저장되어 있다. 최고가와 최저가의 차이를 변동폭으로 정의할 때 (시가 + 변동폭)이 최고가 보다 높을 경우 "상승장", 그렇지 않은 경우 "하락장" 문자열을 출력하라.

> Key Name / Description  
opening_price / 최근 24시간 내 시작 거래금액  
closing_price / 최근 24시간 내 마지막 거래금액  
min_price / 최근 24시간 내 최저 거래금액  
max_price / 최근 24시간 내 최고 거래금액

In [1]:
import requests
btc = requests.get("https://api.bithumb.com/public/ticker/").json()['data']
btc

{'opening_price': '37947000',
 'closing_price': '38061000',
 'min_price': '37760000',
 'max_price': '38282000',
 'units_traded': '771.15283791',
 'acc_trade_value': '29296526955.8362',
 'prev_closing_price': '37951000',
 'units_traded_24H': '1169.78078217',
 'acc_trade_value_24H': '44371998755.3639',
 'fluctate_24H': '290000',
 'fluctate_rate_24H': '0.77',
 'date': '1690437042507'}

In [3]:
변동폭 = float(btc['max_price']) - float(btc['min_price'])
시가 = float(btc['opening_price'])
최고가 = float(btc['max_price'])

if (시가+변동폭) > 최고가:
    print("상승장")
else:
    print("하락장")

상승장


### #131. for문의 실행결과를 예측하라.

In [1]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
    print(변수)
# 사과
# 귤
# 수박

사과
귤
수박


### #132. for문의 실행결과를 예측하라.

In [2]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
  print("#####")
# #####
# #####
# #####

#####
#####
#####


### #133. 다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.

In [3]:
for 변수 in ["A", "B", "C"]:
  print(변수)

A
B
C


In [5]:
# 위의 for문과 동일한 기능을 수행하는 코드
변수 = "A"
print(변수)
변수 = "B"
print(변수)
변수 = "C"
print(변수)

A
B
C


### #134. for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

In [6]:
for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

출력: A
출력: B
출력: C


In [7]:
# 위의 for문과 동일한 기능을 수행하는 코드
변수 = "A"
print("출력:", 변수)
변수 = "B"
print("출력:", 변수)
변수 = "C"
print("출력:", 변수)

출력: A
출력: B
출력: C


### #135. for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

In [8]:
for 변수 in ["A", "B", "C"]:
  b = 변수.lower()
  print("변환:", b)

변환: a
변환: b
변환: c


In [9]:
# 위의 for문과 동일한 기능을 수행하는 코드
변수 = "A"
b = 변수.lower()
print("출력:", b)
변수 = "B"
b = 변수.lower()
print("출력:", b)
변수 = "C"
b = 변수.lower()
print("출력:", b)

출력: a
출력: b
출력: c


### #136. 다음 코드를 for문으로 작성하라.

In [10]:
변수 = 10
print(변수)
변수 = 20
print(변수)
변수 = 30
print(변수)

10
20
30


In [11]:
# for문으로 작성하기
for 변수 in ["10", "20", "30"]:
    print(변수)

10
20
30


### #137. 다음 코드를 for문으로 작성하라.

In [12]:
print(10)
print(20)
print(30)

10
20
30


In [13]:
# for문으로 작성하기
for 변수 in [10, 20, 30]:
    print(변수)

10
20
30


### #138. 다음 코드를 for문으로 작성하라.

In [14]:
print(10)
print("-------")
print(20)
print("-------")
print(30)
print("-------")

10
-------
20
-------
30
-------


In [15]:
# for문으로 작성하기
for 변수 in [10, 20, 30]:
    print(변수)
    print("-------")

10
-------
20
-------
30
-------


### #139. 다음 코드를 for문으로 작성하라.

In [16]:
print("++++")
print(10)
print(20)
print(30)

++++
10
20
30


In [17]:
# for문으로 작성하기
print("++++")
for 변수 in [10, 20, 30]:
    print(변수)

++++
10
20
30


### #140. 다음 코드를 for문으로 작성하라.

In [18]:
print("-------")
print("-------")
print("-------")
print("-------")

-------
-------
-------
-------


In [19]:
# for문으로 작성하기
for i in range(4):
    print("-------")

-------
-------
-------
-------


### #141. 다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

> 리스트 = [100, 200, 300]

In [20]:
리스트 = [100, 200, 300]
for i in 리스트:
    print(i + 10)

110
210
310


### #142. for 문을 사용해서 리스트에 저장된 값을 for문을 사용해서 출력하라.

> 리스트 = ["김밥", "라면", "튀김"]

In [22]:
리스트 = ["김밥", "라면", "튀김"]
for i in 리스트:
    print("오늘의 메뉴:", i)

오늘의 메뉴: 김밥
오늘의 메뉴: 라면
오늘의 메뉴: 튀김


### #143. 리스트에 주식 종목이름이 저장돼 있다.

> 리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
### 저장된 문자열의 길이를 for문을 사용해서 출력하라.

In [23]:
리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
for i in 리스트:
    print(len(i))

6
4
4


### #144. 리스트에는 동물이름이 문자열로 저장돼 있다.

> 리스트 = ['dog', 'cat', 'parrot']
### 동물 이름과 글자수를 다음과 같이 출력하라.

> dog 3  
cat 3  
parrot 6

In [24]:
리스트 = ['dog', 'cat', 'parrot']
for i in 리스트:
    print(i, len(i))

dog 3
cat 3
parrot 6


### #145. 리스트에 동물 이름 저장돼 있다.

> 리스트 = ['dog', 'cat', 'parrot']
### for문을 사용해서 동물 이름의 첫 글자만 출력하라.

In [25]:
리스트 = ['dog', 'cat', 'parrot']
for i in 리스트:
    print(i[0])

d
c
p


### #146. 리스트에는 세 개의 숫자가 바인딩돼 있다.

> 리스트 = [1, 2, 3]
### for문을 사용해서 다음과 같이 출력하라.

>3 x 1  
3 x 2  
3 x 3

In [29]:
리스트 = [1, 2, 3]
for i in 리스트:
    print("3 x", str(i))

3 x 1
3 x 2
3 x 3


### #147. 리스트에는 세 개의 숫자가 바인딩돼 있다.

> 리스트 = [1, 2, 3]
### for문을 사용해서 다음과 같이 출력하라.

>3 x 1 = 3  
3 x 2 = 6  
3 x 3 = 9

In [30]:
리스트 = [1, 2, 3]
for i in 리스트:
    print("3 x", str(i), "=", 3 * i)

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9


### #148. 리스트에는 네 개의 문자열이 바인딩돼 있다.

> 리스트 = ["가", "나", "다", "라"]
### for문을 사용해서 다음과 같이 출력하라.

>나  
다  
라

In [31]:
리스트 = ["가", "나", "다", "라"]
for i in 리스트[1:]:
    print(i)

나
다
라


### #149. 리스트에는 네 개의 문자열이 바인딩돼 있다.

>리스트 = ["가", "나", "다", "라"]
### for문을 사용해서 다음과 같이 출력하라.

>가  
다

In [32]:
리스트 = ["가", "나", "다", "라"]
for i in 리스트[::2]:
    print(i)

가
다


### #150. 리스트에는 네 개의 문자열이 바인딩돼 있다.

> 리스트 = ["가", "나", "다", "라"]
### for문을 사용해서 다음과 같이 출력하라.

>라  
다  
나  
가  

In [33]:
리스트 = ["가", "나", "다", "라"]
for i in 리스트[::-1]:
    print(i)

라
다
나
가
